### ISSUE TITLE
Tvheadend 4.3.20191128

### ISSUE THREAD

**Issue by Modi8200** [SynoCommunity/spksrc#3885]
_For new Package Requests, see the guidelines_  

### Setup
_Package Name:_ Tvheadend
_Package Version:_ 4.3.20191128

_NAS Model:_ DS918+
_NAS Architecture:_ Intel Celeron J3455
_DSM version:_ 6.2.2-24922 update 4

### Expected behavior
Tell us what should happen
There should be EPG for all channels as before upgrade.
### Actual behavior
After upgrade to 4.3 EPG via sat>ip server (digibitR1) stopped working for all English satellite channels. There is only epg for current and next tv program. Picture reception is OK. EPG via hdhomerun works ok.  PLEASE let me know how to install the previous release of TVH which worked perfectly.
Tell us what happens instead

### Steps to reproduce
_1._ 
_2._ 
_3._ 

#### Package log
_Check Package Center or `/usr/local/{package}/var/`_
```
Insert the package log here
```
#### Other logs
_E.g. `/var/log/messages` or `/var/log/synopkg.log`_
```
Insert log here
```

**Comment by th0ma7** [comment]
Hi @Modi8200 ,
You have a few options:
1) manually install the upcoming new version currently in testing and available here: https://github.com/th0ma7/synology/tree/master/packages
2) uninstall current 4.3.x version and manually install the previous 4.2 version instead available here: https://synocommunity.com/package/tvheadend

Also, if you could provide screenshots or additional pointers of the EPG feature you are expecting to work would help understanding the issue is.  

Lastly, personnally I use another method for EPG as documented here, perhaps this will better suit your needs: https://github.com/th0ma7/synology/wiki/TVHeadend:-5)-Guide-t%C3%A9l%C3%A9

**Comment by Modi8200** [comment]
Hi,

Thank you for your quick reply.

In the previous version I used the OTA EPG grabber, it’s easy to configure and there is no additional software required. It provides two weeks EPG for the English satellite channels (DVB-S) and a few days EPG on the Belgian cable (DVB-C) channels.

In release 4.3 the OTA EPG grabber no longer works for DVB-S, the English satellite channels but is still working for cable channels. The option to select  grabbers has disappeared is release 4.3 as you can see in the screendumps

Regards

**Comment by th0ma7** [comment]
Sadly the screen dump didn't show up in your comment on github.  I recommend you use github web interface directly to add the missing info to your comment.

Also, quickly looking around your issue sound similar to the following known tvheadend bug:
* https://tvheadend.org/issues/5769

I would suggest you validate if this is the issue you are facing (and get a few logs to confirm).  If so I may be able to patch something up so the EPG grabber get's packaged as mentioned in the proposed workaround at:
* https://tvheadend.org/issues/5769#note-10

**Comment by Modi8200** [comment]
It looks like my issue is the same as tvh bug 5769. Meanwhile I have downloaded Tvheadend.v16.f15047(apollolake-avaton- but not installed it yet. I am a bit scared of corrupting my Synology should anything go wrong. I took a quick look at the tvh config files with SSH but did not find anything that refers to apolollake or anything else, I saw mention of version 24 though. Here are some pictures :

[pic1.pdf](https://github.com/SynoCommunity/spksrc/files/4213798/pic1.pdf)
[pic2.pdf](https://github.com/SynoCommunity/spksrc/files/4213805/pic2.pdf)
[pic3.pdf](https://github.com/SynoCommunity/spksrc/files/4213807/pic3.pdf)

**Comment by th0ma7** [comment]
This is quite interesting as I'm using the exact same NAS but not facing this issue?
Have you tried the following in order to get a fully fresh install:
- backup your configuration
- remove all instances of tvheadend
- confirm no files still lives into `/var/packages/tvheadend` or `/volume1/@appstore/tvheadend`
- only then reinstall the latest version of the package

**Comment by Modi8200** [comment]
I removed TVH and checked /var/packages and /volume1/@appstore. There was nothing left of TVH. However I noticed that ffmpeg, Node.js_v8 and PHP7.0 are in either directory. Then I reinstalled TVH from the Synocommunity repo but to no avail. The EPG grabber Modules as underscored in pic1 do not show. Unlike release 4.2 release 4.3 does have an option to assign just one grabber in the (dvb-s) muxes but it doesn't do anything.  I also have a hdhomerun (dvb-c) and there the grabber does work automatically. With the same gear (hdhomerun and digibitR1) everything works flawlessly on an ubuntu 18.04 machine. This is the first time I run in problems with TVH. I was quite happy with the previous release which I am considering to reinstall. Do you think I can safely install :  tvheadend.v16.f15047[apollolake-avoton-braswell-broadwell-broadwellnk-bromolow-cedarview-denverton-dockerx64-grantley-kvmx64-x86-x86_64].spk

**Comment by Modi8200** [comment]
After the upgrade to 4.3 I was faced with two issues :
1. My sat2ip server did no longer show under TV adapters although  HDhomerun did and worked well. Initially I suspected a bug in 4.3 but after a while I discovered that the VPN I installed earlier had changed the standard gateway to a value out of reach of TVH. Strangely Release 4.28 didn’t have a problem with that or picked up the correct gateway when the VPN was not yet installed. So when the VPN was disconnected the Sat2ip server was discovered by TVH 4.3 and worked well.
2. Then there was the mystery of the missing EPG grabber modules as shown in  pic1. Without this module one cannot configure the build in OTA grabber. I gave up on release 4.3 and reinstalled 4.28 and the EPG grabber modules were back. Quite coincidentally I changed under General-> Base the default view level (Pic4) from Expert to basic (probably to have smaller windows because with safari on Mac when the window is too large a ghost or shadow window pops up and it’s then impossible to click on cancel for instance, restart the browser is the only way-out .) Further  after setting the default view level to basic I could not believe my eyes that EPG Grabber modules were again missing. So I had finally  found the culprit. I upgrade TVH again to 4.3 and changed the view level setting to expert and the EPG grabber modules were present. So when installing  release 4.28 the standard view level is initially set to expert and on 4.3 to basic ? BTW, TVH on my Ubuntu system is set on user interface level Basic and the EPG grabber is present. Anyway the problem is solved, no real bugs found but small things sometimes have big consequences. Thanks for your help.
[pic4.pdf](https://github.com/SynoCommunity/spksrc/files/4250764/pic4.pdf)
[pic5_shadow.pdf](https://github.com/SynoCommunity/spksrc/files/4250766/pic5_shadow.pdf)
